src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1139,13 +1141,45 @@ private void CheckRestrictedTypeReceiver(BoundExpression expression, CSharpCompi
                 }
             }
 
-            return new BoundCall(node, receiver, method, args, argNames, argRefKinds, isDelegateCall: isDelegateCall,
+            return new BoundCall(node, receiver, initialBindingReceiverIsSubjectToCloning: ReceiverIsSubjectToCloning(receiver, method), method, args, argNames, argRefKinds, isDelegateCall: isDelegateCall,
                         expanded: expanded, invokedAsExtensionMethod: invokedAsExtensionMethod,
                         argsToParamsOpt: argsToParams, defaultArguments, resultKind: LookupResultKind.Viable, type: returnType, hasErrors: gotError);
         }
 
 #nullable enable
 
+        internal ThreeState ReceiverIsSubjectToCloning(BoundExpression? receiver, PropertySymbol property)
+            => ReceiverIsSubjectToCloning(receiver, property.GetMethod ?? property.SetMethod);
+
+        private static bool ReceiverCannotBeSubjectToCloning([NotNullWhen(returnValue: false)] BoundExpression? receiver)
+        {
+            return receiver is BoundValuePlaceholderBase || receiver?.Type?.IsValueType != true;
+        }
+
+        internal ThreeState ReceiverIsSubjectToCloning(BoundExpression? receiver, MethodSymbol method)
+        {
+            if (ReceiverCannotBeSubjectToCloning(receiver))
+            {
+                return ThreeState.False;
+            }
+
+            var valueKind = method.IsEffectivelyReadOnly
+                ? BindValueKind.RefersToLocation
+                : BindValueKind.RefersToLocation | BindValueKind.Assignable;
+            var result = !CheckValueKind(receiver.Syntax, receiver, valueKind, checkingReceiver: true, BindingDiagnosticBag.Discarded);
+            return result.ToThreeState();
+        }
+
+        internal static ThreeState TryReceiverIsSubjectToCloning(BoundExpression? receiver)
```

Comment:
>TryReceiverIsSubjectToCloning [](http://example.com/codeflow?start=35&length=29)

Based on the call sites, it looks like we should be able to eliminate this helper.